### PR TITLE
[MAG-3115] Asset meta component

### PR DESCRIPTION
**Pull request by matejcsok-istvan**
Build the Asset Meta components

## TL;DR

-   FeatureList component added to Molecules
-   AssetMeta component added to Organisms

## Details

Checklist / Reminders:

-   [ ] Documentation / Story is created / adapted
-   [ ] Unit tests
-   [ ] CSS changes are BEM-compliant

FeatureList component:
Screenshot 2020-12-09 at 19 28 08

AssetMeta component:
Screenshot 2020-12-09 at 19 28 20

### CODE REVIEW

File: src/components/Molecules/FeatureList/FeatureList.tsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+const ListWrapper = styled.div`
+  padding-top: ${({ theme }) => theme.spacing.base.sm};
+  padding-bottom: ${({ theme }) => theme.spacing.base.sm};
+`
+
+export const FeatureList: React.FC<FeatureListProps> = ({ list }): JSX.Element => {
+  return (
+    <FlexBox display="flex" flexDirection="column">
+      <Divider color="grey1" height={2} />
+      {list.map(item => (
```
Comment (matthill8286) [inline src/components/Molecules/FeatureList/FeatureList.tsx:23]: Could you cover if this component is waiting for data and renders without the list and also if the list  isn't an array. Just to cover misuse 👍

File: src/components/Organisms/AssetMeta/AssetMeta.test.tsx [hunk 2]
```diff
@@ -0,0 +1,32 @@
+import React from 'react'
+import { renderWithTheme } from '@/testRenderer'
+import { AssetMeta } from './AssetMeta'
+
+describe('Asset page meta', () => {
+  it('renders asset meta', () => {
+    const wrapper = renderWithTheme(
+      <AssetMeta
+        isShareInteraction={true}
```
Comment (matthill8286) [inline src/components/Organisms/AssetMeta/AssetMeta.test.tsx:9]:
```suggestion
        isShareInteraction
```

File: src/components/Organisms/AssetMeta/AssetMeta.test.tsx [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+        isBookmarkInteraction={true}
```
Comment (matthill8286) [inline src/components/Organisms/AssetMeta/AssetMeta.test.tsx:10]:
```suggestion
        isBookmarkInteraction
```

File: src/components/Organisms/AssetMeta/AssetMeta.test.tsx [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+        isCreatePlaylistInteraction={true}
```
Comment (matthill8286) [inline src/components/Organisms/AssetMeta/AssetMeta.test.tsx:11]:
```suggestion
        isCreatePlaylistInteraction
```

File: src/components/Organisms/AssetMeta/AssetMeta.tsx [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+            <Icon>
+              <SvgSaveToPlaylist />
+            </Icon>
+          </FlexItem>
+        )}
+      </IconsWrapper>
+      <Card
+        elevation={1}
+        shape="rounded-small"
+        padding={{
```
Comment (matthill8286) [inline src/components/Organisms/AssetMeta/AssetMeta.tsx:65]:
```suggestion
        padding="lg"
```

**Review by matthill8286 — APPROVED** [review]
A few comments. Overall good.

**Comment by matthill8286** [comment]
@matejcsok-istvan could you also add screenshots of your work please. Thanks.

File: src/components/Molecules/FeatureList/FeatureList.tsx [hunk 6]
```diff
@@ -0,0 +1,48 @@
+import React from 'react'
+import { Divider } from '@/components/Atoms/Divider'
+import { CopyText } from '@/components/Atoms/Typography'
+import { FlexBox } from '@/components/Helper/FlexBox'
+import { Cell, Grid, Row } from '@/components/Helper/Grid'
+import { styled } from '@/styles'
+
+export type FeatureListType = { label: string; value: string }[]
```
Comment (matthill8286) [inline src/components/Molecules/FeatureList/FeatureList.tsx:8]:
```suggestion
export type FeatureListType = { label: string; value: string }
```

File: src/components/Molecules/FeatureList/FeatureList.tsx [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+
+export interface FeatureListProps {
+  list: FeatureListType
```
Comment (matthill8286) [inline src/components/Molecules/FeatureList/FeatureList.tsx:11]:
```suggestion
  list: FeatureListType[]
```

File: src/components/Organisms/AssetMeta/AssetMeta.tsx [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+  buttonTextComplete: string
+  list: FeatureListType
+  competencyCopy: string
+  tags: { label: string; href: string }[]
+}
+
+export const AssetMeta: React.FC<AssetMetaProps> = ({
+  isShareInteraction,
+  isBookmarkInteraction,
+  isCreatePlaylistInteraction,
```
Comment (matthill8286) [inline src/components/Organisms/AssetMeta/AssetMeta.tsx:30]: We can just default these too true, and control which we want to turn off if we ever do but a single override.

File: src/components/Organisms/AssetMeta/AssetMeta.tsx [hunk 9]
```diff
@@ -0,0 +1,104 @@
...
+
+export interface AssetMetaProps {
+  isShareInteraction: boolean
+  isBookmarkInteraction: boolean
+  isCreatePlaylistInteraction: boolean
+  buttonTextLanch: string
+  buttonTextComplete: string
+  list: FeatureListType
+  competencyCopy: string
+  tags: { label: string; href: string }[]
```
Comment (matthill8286) [inline src/components/Organisms/AssetMeta/AssetMeta.tsx:24]:
```suggestion
  tags: Partial<TagProps>[]
```

File: src/components/Organisms/AssetMeta/AssetMeta.tsx [hunk 10]
```diff
@@ -0,0 +1,104 @@
...
+        <CompetencyText color="grey4">{competencyCopy}</CompetencyText>
+        <FlexBox flexWrap="wrap">
+          {tags.map((tag, index: number) => (
+            <FlexItem key={index}>
+              <Tag
+                curved
+                color="primary"
+                marginBottom
+                marginRight
+                text={tag.label}
```
Comment (matthill8286) [inline src/components/Organisms/AssetMeta/AssetMeta.tsx:93]: Would be good to keep the types synced, as TagProps is expecting the tag to have a text or href
```suggestion
                text={tag.text}
```

File: src/components/Atoms/Tag/Tag.interface.ts [hunk 11]
```diff
@@ -8,13 +8,15 @@ export interface TagProps {
   marginBottom?: boolean
   color?: ThemeColors
   className?: string
+  curved?: boolean
```
Comment (matthill8286) [inline src/components/Atoms/Tag/Tag.interface.ts:11]: By default just make the tags curved as thats what the designs dictate.
```suggestion
```

File: src/components/Atoms/Tag/Tag.interface.ts [hunk 12]
```diff
@@ -8,13 +8,15 @@ export interface TagProps {
...
 }
 
 export interface StyledTagWrapperProps {
   hasLink?: boolean
   marginRight?: boolean
   marginBottom?: boolean
   color?: ThemeColors
+  curved?: boolean
```
Comment (matthill8286) [inline src/components/Atoms/Tag/Tag.interface.ts:19]:
```suggestion
```

File: src/components/Atoms/Tag/Tag.styled.tsx [hunk 13]
```diff
@@ -3,12 +3,12 @@ import { Theme } from '@/types'
...
 
 export const StyledTagWrapper = styled.div<StyledTagWrapperProps>(
-  ({ theme, hasLink, marginRight, color }) => css`
+  ({ theme, hasLink, marginRight, color, curved }) => css`
     color: ${theme.color.grey5};
     padding: ${theme.spacing.base.xxs} ${theme.spacing.base.xs};
     border-color: ${!color ? theme.color.grey2 : theme.color[color]};
     border-width: 1px;
-    border-radius: ${theme.dimension.borderRadius1};
+    border-radius: ${curved ? theme.dimension.borderRadius6 : theme.dimension.borderRadius1};
```
Comment (matthill8286) [inline src/components/Atoms/Tag/Tag.styled.tsx:11]:
```suggestion
    border-radius: ${theme.dimension.borderRadius6};
```

File: src/components/Atoms/Tag/Tag.styled.tsx [hunk 14]
```diff
@@ -3,12 +3,12 @@ import { Theme } from '@/types'
 import { StyledTagWrapperProps } from './Tag.interface'
 
 export const StyledTagWrapper = styled.div<StyledTagWrapperProps>(
-  ({ theme, hasLink, marginRight, color }) => css`
+  ({ theme, hasLink, marginRight, color, curved }) => css`
```
Comment (matthill8286) [inline src/components/Atoms/Tag/Tag.styled.tsx:6]:
```suggestion
  ({ theme, hasLink, marginRight, color }) => css`
```

File: src/components/Atoms/Tag/Tag.tsx [hunk 15]
```diff
@@ -23,6 +23,7 @@ export const Tag: React.FC<TagProps> = ({
   marginBottom,
   marginRight,
   color,
+  curved,
```
Comment (matthill8286) [inline src/components/Atoms/Tag/Tag.tsx:26]:
```suggestion
```

File: src/components/Atoms/Tag/Tag.tsx [hunk 16]
```diff
@@ -31,7 +32,8 @@ export const Tag: React.FC<TagProps> = ({
       marginBottom={marginBottom}
       marginRight={marginRight}
       hasLink={!!href}
-      color={color}>
+      color={color}
+      curved={curved}>
```
Comment (matthill8286) [inline src/components/Atoms/Tag/Tag.tsx:36]:
```suggestion
      >
```

File: src/components/Molecules/FeatureList/FeatureList.story.tsx [hunk 17]
```diff
@@ -0,0 +1,18 @@
+import { storiesOf } from '@storybook/react'
+import * as React from 'react'
+import { FeatureList } from './FeatureList'
+
+const componentProps = {
```
Comment (matthill8286) [inline src/components/Molecules/FeatureList/FeatureList.story.tsx:5]:
```suggestion
export const featureListProps = {
```

File: src/components/Molecules/FeatureList/FeatureList.story.tsx [hunk 18]
```diff
@@ -0,0 +1,18 @@
...
+    { label: 'Provider', value: 'Youtube' },
+    { label: 'Primary competency', value: 'Digital Marketing' },
+    { label: 'Type', value: 'Article' },
+    { label: 'Length', value: '16 mins' },
+  ],
+}
+
+storiesOf('Design System/Molecules/FeatureList', module).add('Default', () => (
+  <div style={{ width: '300px' }}>
+    <FeatureList {...componentProps} />
```
Comment (matthill8286) [inline src/components/Molecules/FeatureList/FeatureList.story.tsx:16]:
```suggestion
    <FeatureList {...featureListProps} />
```

File: src/components/Molecules/FeatureList/FeatureList.test.tsx [hunk 19]
```diff
@@ -0,0 +1,19 @@
+import React from 'react'
+import { renderWithTheme } from '@/testRenderer'
+import { FeatureList } from './FeatureList'
+
+describe('Meta Feature list', () => {
+  it('renders feature list', () => {
+    const wrapper = renderWithTheme(
+      <FeatureList
+        list={[
```
Comment (matthill8286) [inline src/components/Molecules/FeatureList/FeatureList.test.tsx:9]:
```suggestion
        list={[...featureListProps.list]}
```

**Comment by matthill8286** [comment]
@matejcsok-istvan After seeing the screenshots, a few things, concentrate on reusing types/interfaces

- the top icons have a rounded border, the 3rd icon looks smaller
- the icons in the primary button is white and not grey
- the asset list the left values are bolder than the right values
- the tags look to have too much vertical padding and they need to also be bold

I have attached a screenshot for reference, on a further note about  the top icons, perhaps create a separate molecule so we can control a theme/colors if these need to be reused elsewhere.

Screenshot 2020-12-10 at 10 18 57

**Comment by matejcsok-istvan** [comment]
I think I fixed all the issues except for the icons, the second icon overwrites the next two icons, I think the this is the problem: [stackoverflow](https://stackoverflow.com/questions/47359136/react-rendering-svg-overwrites-other-svgs-on-the-page), but I am not sure how to solve it, I spent some time on it, but I couldn't figure it out, maybe you know something I don't, I can probably find a solution but it takes more time
.
New screenshots:
FeatureList loading state:
Screenshot 2020-12-15 at 9 43 55

FeatureList:
Screenshot 2020-12-15 at 9 46 36

AssetMeta component:
Screenshot 2020-12-15 at 9 45 53

**Comment by matthill8286** [comment]
@matejcsok-istvan all good, the only thing is remove the loading state as it looks awful. It's meant to mimic the layout almost like optimistic ui, so it'd be very slender skeletons for each piece of text. Just an FYI. So once thats removed it's good to go. Don't worry about the icons as i'm building a SVG to JSX library and this will solve any icon issues.

**Review by matthill8286 — APPROVED** [review]

**Pull request merged**